.github/workflows/soak-testing.yml

```diff
@@ -0,0 +1,237 @@
+# Pre-requisites:
+# - AWS Account with correct permissions
+# - `gh-pages` branch
+# - AWS Account needs to add the LOG_GROUP_NAME defined below
+
+name: Soak Testing
+on:
+  workflow_dispatch:
+    inputs:
+      target_commit_sha:
+        description: 'The commit SHA on this repo to use for the Soak Tests.'
+        required: true
+      test_duration_minutes:
+        description: 'The duration of the Soak Tests in minutes.'
+        required: true
+        default: 300
+  schedule:
+    - cron: '0 15 * * *'
+env:
+  # NOTE: The configuration of `APP_PROCESS_EXECUTABLE_NAME` is repo dependent
+  APP_PROCESS_EXECUTABLE_NAME: EXAMPLE_OF_SOMETHING
+  AWS_DEFAULT_REGION: us-east-1
+  DEFAULT_TEST_DURATION_MINUTES: 300
+  HOSTMETRICS_INTERVAL_SECS: 600
+  CPU_LOAD_THRESHOLD: 50
+  TOTAL_MEMORY_THRESHOLD: 1073741824 # 1 GiB
+  MAX_BENCHMARKS_TO_KEEP: 100
+  LISTEN_ADDRESS_PORT: 8080
+  # TODO: We might be able to adapt the "Soak Tests" to be "Overhead Tests".
+  # This means monitoring the Sample App's performance using high levels of TPS
+  # for the Load Generator over a shorter period of testing time. For example:
+  # https://github.com/aws-observability/aws-otel-collector/blob/main/docs/performance_model.md
+  # THROUGHPUT_PER_SECOND: TBD?
+
+jobs:
+  test_apps_and_publish_results:
+    name: Soak Performance Test - (${{ matrix.app-platform }}, ${{ matrix.instrumentation-type }})
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      id-token: write
+      issues: write
+    strategy:
+      fail-fast: false
+      matrix:
+        app-platform: [ rails ]
```

Comment:
should we follow [this](https://github.com/NathanielRN/aws-otel-ruby/pull/1/files#diff-23839e1cff0074a71eae9881bf8375ffbafc1fc6a29bf9a1cb4487b2c6306ef0L25)

Reply:
Same as above.